### PR TITLE
Add BTPayPalVaultRequest

### PR DESCRIPTION
**Pull request by sestevens**
### Summary of changes

- Add `BTPayPalVaultRequest`

_Follow-up PR will include:_
- Deprecating `requestBillingAgreement` and `requestOneTimePayment` methods
- Adding `tokenizePayPalAccount` method
- Updating the migration guide

### Checklist

- [x] Added a changelog entry

- @sestevens

### CODE REVIEW

**Review by scannillo — APPROVED** [review]
🏁

File: Sources/BraintreePayPal/BTPayPalRequest_Internal.h [hunk 1]
```diff
@@ -11,11 +11,18 @@ NS_ASSUME_NONNULL_BEGIN
 extern NSString *const BTPayPalCallbackURLHostAndPath;
 extern NSString *const BTPayPalCallbackURLScheme;
 
+typedef NS_ENUM(NSUInteger, BTPayPalPaymentType) {
+    BTPayPalPaymentTypeCheckout,
+    BTPayPalPaymentTypeBillingAgreement
```
Comment (sarahkoop) [inline Sources/BraintreePayPal/BTPayPalRequest_Internal.h:16]: Maybe we should rename this to `BTPayPalPaymentTypeVault` to align with the request type name

**Review by sarahkoop — APPROVED** [review]
🌟

File: Demo/Application/Features/PayPal - Credit/BraintreeDemoPayPalCreditPaymentViewController.m [hunk 2]
```diff
@@ -49,12 +49,12 @@ - (void)tappedPayPalOneTimePayment:(UIButton *)sender {
     [sender setEnabled:NO];
 
     BTPayPalDriver *driver = [[BTPayPalDriver alloc] initWithAPIClient:self.apiClient];
-    BTPayPalCheckoutRequest *request = [[BTPayPalCheckoutRequest alloc] initWithAmount:@"4.30"];
-    request.activeWindow = self.view.window;
-
-    request.offerCredit = YES;
 
     if (self.paypalTypeSwitch.selectedSegmentIndex == 0) {
+        BTPayPalCheckoutRequest *request = [[BTPayPalCheckoutRequest alloc] initWithAmount:@"4.30"];
```
Comment (hollabaq86) [inline Demo/Application/Features/PayPal - Credit/BraintreeDemoPayPalCreditPaymentViewController.m:54]: I wonder if we should rename this to `BRaintreeDemoPayPalPayLaterPaymentViewController`?

File: Sources/BraintreePayPal/BTPayPalDriver.m [hunk 3]
```diff
@@ -94,27 +94,22 @@ - (void)dealloc {
...
                    completion:(void (^)(BTPayPalAccountNonce *tokenizedCheckout, NSError *error))completionBlock {
-    [self requestPayPalCheckout:request
-             isBillingAgreement:NO
-                     completion:completionBlock];
+    [self requestPayPalCheckout:request completion:completionBlock];
 }
 
 #pragma mark - Helpers
 
 /// A "Hermes checkout" is used by both Billing Agreements (Vault) and One-Time Payments (Checkout)
```
Comment (hollabaq86) [inline Sources/BraintreePayPal/BTPayPalDriver.m:111]:
```suggestion
/// Used by both Billing Agreements (Vault) and One-Time Payments (Checkout)
```

Not sure if we need to mention `Hermes checkout` here specifically, just documenting how this is used should be fine

File: Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalDriver.h [hunk 4]
```diff
@@ -109,10 +111,10 @@ typedef NS_ENUM(NSInteger, BTPayPalDriverErrorType) {
  
  On success, you will receive an instance of `BTPayPalAccountNonce`; on failure or user cancelation you will receive an error. If the user cancels out of the flow, the error code will be `BTPayPalDriverErrorTypeCanceled`.
 
- @param request A PayPal request
+ @param request A PayPal vault request
```
Comment (hollabaq86) [inline Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalDriver.h:114]:
```suggestion
 @param request A PayPal Vault request
```

File: Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalDriver.h [hunk 5]
```diff
@@ -95,7 +97,7 @@ typedef NS_ENUM(NSInteger, BTPayPalDriverErrorType) {
 
  On success, you will receive an instance of `BTPayPalAccountNonce`; on failure or user cancelation you will receive an error. If the user cancels out of the flow, the error code will be `BTPayPalDriverErrorTypeCanceled`.
 
- @param request A PayPal request
+ @param request A PayPal checkout request
```
Comment (hollabaq86) [inline Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalDriver.h:100]:
```suggestion
 @param request A PayPal Checkout request
```

Comment (hollabaq86) [reply Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalDriver.h:100]: this is a super small nit but I _think_ our dev docs capitalize `Checkout` and `Vault` flows

File: Sources/BraintreePayPal/BTPayPalDriver.m [hunk 6]
```diff
@@ -177,15 +169,12 @@ - (void)requestPayPalCheckout:(BTPayPalRequest *)request
...
-            } else {
-                [self sendAnalyticsEventForInitiatingOneTouchForPaymentType:BTPayPalPaymentTypeCheckout withSuccess:analyticsSuccess];
-            }
+
+            [self sendAnalyticsEventForInitiatingOneTouchForPaymentType:request.paymentType withSuccess:analyticsSuccess];
 
             [self handlePayPalRequestWithURL:approvalUrl
                                        error:error
-                                 paymentType:isBillingAgreement ? BTPayPalPaymentTypeBillingAgreement : BTPayPalPaymentTypeCheckout
+                                 paymentType:request.paymentType
```
Comment (sshropshire) [inline Sources/BraintreePayPal/BTPayPalDriver.m:177]: 🙌

File: Sources/BraintreePayPal/BTPayPalDriver.m (hunk 3 above)
Comment (sestevens) [reply Sources/BraintreePayPal/BTPayPalDriver.m:111]: Yeah, this comment isn't super helpful. In the next PR, this method is made public and renamed to `tokenizePayPalAccount`, and we delete this comment alongside that change. So I'm just going to leave it as-is here, knowing that it'll be deleted when the next PR is merged.

**Review by sshropshire — APPROVED** [review]

File: Sources/BraintreePayPal/BTPayPalRequest_Internal.h (hunk 1 above)
Comment (sestevens) [reply Sources/BraintreePayPal/BTPayPalRequest_Internal.h]: Good call! Updated.

File: Demo/Application/Features/PayPal - Credit/BraintreeDemoPayPalCreditPaymentViewController.m (hunk 2 above)
Comment (sestevens) [reply Demo/Application/Features/PayPal - Credit/BraintreeDemoPayPalCreditPaymentViewController.m:54]: Maybe let's save that for a later PR.

File: Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalDriver.h (hunk 4 above)
Comment (sestevens) [reply Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalDriver.h]: Updated!

File: Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalDriver.h (hunk 5 above)
Comment (sestevens) [reply Sources/BraintreePayPal/Public/BraintreePayPal/BTPayPalDriver.h]: Updated!

**Pull request merged**